### DISCUSSION TITLE
Breakpoint text size VS line-height

### DISCUSSION THREAD

**Discussion by frank-laemmer** [tailwindlabs/tailwindcss#12291] [Help]
**What version of Tailwind CSS are you using?**

v3.3.5

**Reproduction URL**

https://play.tailwindcss.com/wfrFmp1hfs

<img width="1401" alt="Screenshot 2023-10-25 at 21 27 53" src="https://github.com/tailwindlabs/tailwindcss/assets/962736/e5f12eef-7b50-4c14-9af4-431920849e18">

**Describe your issue**

I apply a text modifier (breakpoint prefix) to text size. I expect that this will alter the text size on certain screen sizes but not touch the line-height. My goal is something like this:

```html
<p class="text-lg xl:text-2xl 3xl:text-4xl leading-relaxed">Lorem ispum</p>
```
In other words, I'd like to change the text size with different screens, the leading should remain. 

This may work (not tried), but it doesn't feel right to me.

```html
<p class="text-lg xl:text-2xl 3xl:text-4xl leading-relaxed xl:leading-relaxed 3xl:leading-relaxed">Lorem ispum</p>
```

I think the 'issue' is this line-height definition. The media query makes this rule more specific than the other one.

```css
@media (min-width: 640px) {
  .sm\:text-xl {
    font-size: 1.25rem;
    line-height: 1.75rem;
  }
}
```

Maybe I am missing something? Is this a feature?

**Comment by thecrypticace** [comment]
The decision to include line height when changing the text size is by design as it's a better default for most use cases and you can override it if you want.

If you want to override the line height at a breakpoint you can do one of two things:
- Set a specific line height at the same time as the text size: `sm:text-xl/relaxed`
- Change the line height at the same breakpoint: `sm:text-xl sm:leading-relaxed`

**Reply by adamwathan** [reply]
In addition to what @thecrypticace suggested, you can also update your own Tailwind config to not include line heights with your font sizes if you want to avoid ever assigning a line-height automatically when assigning a font size.

**Reply by frank-laemmer** [reply]
Thanks for quick replies and suggestions. I understand that line-height is a good candidate for opinionated defaults to make websites look nice. Web developers don't bother, web designers do.

Yet from Tailwind as a utility framework I would not have expected that behavior. My understanding of such a system is that it's explicit and not implicit. When I set the text-size for a container query, I only want that and nothing else here.

I will explore tinkering with the Tailwind config. You may close this. Thanks again.
